logback/src/main/java/com/newrelic/logging/logback/JsonFactoryProvider.java

```diff
@@ -0,0 +1,18 @@
+package com.newrelic.logging.logback;
+
+import com.fasterxml.jackson.core.JsonFactory;
+
+public class JsonFactoryProvider {
+    private static JsonFactory jsonFactory;
+
+    static JsonFactory getInstance() {
+        if (jsonFactory == null) {
```

Comment:
So it might be a bit pedantic and may not seriously impact real-world applications, but technically there is a race condition here. 

Imagine 10 threads, all calling `getInstance()` at the same time.  It is possible for all 10 to pass the null check, then they all create instances and assign them and return them.  For some period of time, there will be more than 1 instance of `JsonFactory` around.

In this case, especially since `JsonFactory` has no dependencies and is so easy to construct, I think it is easily remedied by changing the static field declaration to:

```
    private final static JsonFactory jsonFactory = new JsonFactory();

```
and then you can remove the `if()` in the getter.  Should be safer and simpler that way.

Reply:
Good catch! I'll change it accordingly.

Reply:
Done in cb71c26c66b5eaf495b2db0d9fcc36463d9c10ad